src/avr/avr_m328p.c

```diff
@@ -259,6 +266,52 @@ static void timer0_normal(struct MSIM_AVR *mcu,
 	}
 }
 
+static void timer0_ctc(struct MSIM_AVR *mcu,
+                          uint32_t presc, uint32_t *ticks,
+                          uint8_t wgm0, uint8_t com0a, uint8_t com0b)
+{
+	uint8_t top = DM(OCR0A);
+
+	switch(wgm0)
+	if ((*ticks) < (presc-1U)) {
+		(*ticks)++;
+	} else if ((*ticks) > (presc-1U)) {
+		snprintf(mcu->log, sizeof mcu->log, "number of Timer1 "
+		         "ticks=%" PRIu32 " should be <= (prescaler-1)=%"
+		         PRIu32 "; timer1 will not be updated!",
+		         *ticks, (presc-1U));
+		MSIM_LOG_ERROR(mcu->log);
+	} else {
+		if (DM(TCNT0) == max_8bit || DM(TCNT0) == top) {
+			/* Generate TOV flag always and only on 0xFF */
+			if(DM(TCNT0) == max_8bit) {
+				/* Set TimerCounter0 overflow flag  */
+				DM(TIFR0) |= DM(1<<TOV0);
+			}
+			/* Generate Interrupt on TOP value */
+			if (DM(TCNT0) == DM(OCR0A)) {
+				/* Set TC0 Output Compare A Flag */
+				DM(TIFR) |= (1<<OCF0A);
```

Comment:
ATmega328 doesn't have a TIFR register. There are three actually: TIFR0, TIFR1 and TIFR2. You need the first one, TIFR0 to set OCF0A flag.